api/lambdas/sqs-to-converter/index.js

```diff
@@ -47,11 +56,90 @@ const fhirConverter = axios.create({
   },
 });
 
+function isSidechainConnector() {
+  return sidechainFHIRConverterUrl ? true : false;
+}
+
+async function getSidechainConverterAPIKey() {
+  if (!sidechainFHIRConverterKeysSecretName) {
+    throw new Error(`Programming error - sidechainFHIRConverterKeys is not set`);
+  }
+
+  const secret = await getSecret(sidechainFHIRConverterKeysSecretName);
+  if (!secret) {
+    throw new Error(`Config error - sidechainFHIRConverterKeysSecret doesn't exist`);
+  }
+
+  const keys = secret.split(",");
+  if (keys.length < 1) {
+    throw new Error(
+      `Config error - sidechainFHIRConverterKeysSecret needs to be a comma separated string of keys`
+    );
+  }
+  // pick a key at random
+  return keys[Math.floor(Math.random() * keys.length)];
+}
+
+function postProcessSidechainFHIRBundle(fhirBundle, extension) {
+  fhirBundle.type = "batch";
+
+  const stringsToReplace = [];
+  if (fhirBundle?.entry?.length) {
+    for (const bundleEntry of fhirBundle.entry) {
+      // add doc id extension
+      if (!bundleEntry.resource.extension) bundleEntry.resource.extension = [];
+      bundleEntry.resource.extension.push(extension);
+
+      // validate resource id
+      let idToUse = bundleEntry.resource.id;
+      if (!uuid.validate(!idToUse)) {
+        // if it's not valid, we'll need to generate a valid UUID
+        const newId = uuid.v4();
+        bundleEntry.resource.id = newId;
+
+        // save the old/new ID pair so we later replace all occurences
+        // of the old one with the new one
+        stringsToReplace.push({ old: idToUse, new: newId });
+
+        idToUse = newId;
+      }
+
+      // change the fullUrl in the resource to match what our converter would generate
+      bundleEntry.fullUrl = `urn:uuid:${idToUse}`;
```

Comment:
Sanity check: this is not removing the traceability from other other refs to this one, right?

Reply:
no it's not - it's outside of any resource body

FYI the sidechain converter will give us a URL like `https://api.domain.com/Patient/054d186c-1d80-40b2-ba75-9c1f6b21ff16` by default here

Reply:
> the sidechain converter will give us a URL like https://api.domain.com/Patient/054d186c-1d80-40b2-ba75-9c1f6b21ff16 by default here

Yeah, that's what I mean. Other resources referencing a Patient with the ID `https://api.domain.com/Patient/054d186c-1d80-40b2-ba75-9c1f6b21ff16` will lose that internal Bundle reference. While it might be pointing to the URL of the to-be-created Patient, the FHIR server might not be able to build a dependency tree properly.

Reply:
Discussed offline, this won't be an issue since the references to the resource from other places will be `Patient/054d186c-1d80-40b2-ba75-9c1f6b21ff16` only (without the url/domain part)